src/subapps/projects/hooks/useStepStatus.ts

```diff
@@ -0,0 +1,39 @@
+import * as React from 'react';
+import { useNexusContext } from '@bbp/react-nexus';
+
+import fusionConfig from '../config';
+
+export const useStepStatus = (orgLabel: string, projectLabel: string) => {
+  const nexus = useNexusContext();
+  const [error, setError] = React.useState<Error>();
+  const [success, setSuccess] = React.useState<boolean>();
+
+  const updateStatus = async (
+    stepId: string,
```

Comment:
```
try {

const source = await nexus.Resource.getSource(
      orgLabel,
      projectLabel,
      encodeURIComponent(stepId)
    );
const updateResponse = await nexus.Resource.update(orgLabel, projectLabel, stepId, rev, {
          ...originalPayload,
          status: newStatus,
          '@type': fusionConfig.workflowStepType,
        });
}
if(updateResonse) {
setSuccess(true)
}
catch (error) {
setError(error)
}
```

Reply:
Some thing like the above would have been better. We should try not to mix the "then" style code with "await".